### ISSUE TITLE
Could not evaluate: no implicit conversion of String into Integer

### ISSUE THREAD

**Issue by pjmorr** [puppetlabs/control-repo#10]
On the puppetmaster I receive this error during every run.
Granted, this is PE2015.3 and you mentioned it wasn't tested, but I have been able to get everything working using the native Code Management system with the exception of removing these two messages.
Error: /Stage[main]/Profile::Puppetmaster/Git_deploy_key[add_deploy_key_to_puppet_control-<redacted>]: Could not evaluate: no implicit conversion of String into Integer
Error: /Stage[main]/Profile::Puppetmaster/Git_webhook[web_post_receive_webhook-<redacted>]: Could not evaluate: no implicit conversion of String into Integer

Also, the webhook and deploy key were put in place manually and are working just fine, I just can't seem to get this error to go away.

**Comment by npwalker** [comment]
@pjmorr I believe this issue is either a permissions issue with the r10k_api_user not being able to add / modify the deploy keys and webhooks on your control repo.  Or that the control repo is not called puppet/control-repo.  

The errors you are getting come from abrader-gms type and provider and I thought I was using a version that should have better error messages but maybe I need to update that.  You could try updating the abrader-gms module in your Puppetfile to use a newer commit.  CC @abrader 

Also, are you using gitlab or are you using a different Git management system?

**Comment by pjmorr** [comment]
sorry for the delay, I am using Gilab and I do believe you are correct.  After some digging I sorted things out and do believe I have everything running great on PE 2015.3.1 even...  Between this, trusted node data and Application Orchestration I am loosing the battle to not make the evil cackle of automation bliss!

Turns out I had erroneously placed a value in place of the , undef ) which was causing my grief.

Thanks!
